test/integration_strict_test.go

```diff
@@ -99,27 +92,23 @@ func TestRootTerragruntHCLStrictMode(t *testing.T) {
 		{
 			name:           "root terragrunt.hcl",
 			strictMode:     false,
-			expectedStderr: strict.StrictControls[strict.RootTerragruntHCL].Warning,
+			expectedStderr: "Using `terragrunt.hcl` as the root of Terragrunt configurations is an anti-pattern",
 		},
 		{
 			name:          "root terragrunt.hcl with root-terragrunt-hcl strict control",
 			controls:      []string{"root-terragrunt-hcl"},
 			strictMode:    false,
-			expectedError: strict.StrictControls[strict.RootTerragruntHCL].Error,
-		},
-		{
-			name:          "root terragrunt.hcl with strict mode",
-			strictMode:    true,
-			expectedError: strict.StrictControls[strict.RootTerragruntHCL].Error,
+			expectedError: errors.New("Using `terragrunt.hcl` as the root of Terragrunt configurations is an anti-pattern"),
 		},
+		// we cannot test `-strict-mode` flag, since we cannot know at which strict control TG will output the error.
```

Comment:
I don't understand this.

Reply:
After adding a new strict control or ending an old one, using `-strict-mode` will return different errors, you will always have to edit this test. When map structure was used, the order was not guaranteed and the error could always be different, even without adding or removing a new control.